python/semantic_kernel/connectors/ai/onnx/services/onnx_gen_ai_chat_completion.py

```
@@ -0,0 +1,184 @@
+                StreamingChatMessageContent(
+                    role=AuthorRole.ASSISTANT,
+                    choice_index=0,
+                    content=new_token,
+                    ai_model_id=self.ai_model_id,
+                )
+            ]
+
+    async def _apply_chat_template(self, chat_history: "ChatHistory", **kwargs) -> str:
+        kernel = kwargs.get("kernel", None)
+        arguments = kwargs.get("arguments", None)
+        if kernel is None:
+            raise ServiceInvalidExecutionSettingsError("The kernel is required for OnnxChatCompletion")
+        if arguments is None:
+            arguments = KernelArguments(messages=self._prepare_chat_history_for_request(chat_history))
+            logger.warning("The arguments were not provided, auto injecting chat history as messages.")
+        template = TEMPLATE_FORMAT_MAP[self.prompt_template.template_format](
+            prompt_template_config=self.prompt_template
+        )  # type: ignore
+        return await template.render(kernel, arguments)
+
+    async def _get_images_from_history(self, chat_history: "ChatHistory") -> ImageContent | None:
+        images = []
+        for message in chat_history.messages:
+            image_content = message.items
+            for image in image_content:
+                if isinstance(image, ImageContent):
+                    if not self.enable_multi_modality:
+                        raise ServiceInvalidExecutionSettingsError("The model does not support multi-modality")
+                    if image.uri:
+                        images.append(image)
+                    else:
+                        raise ServiceInvalidExecutionSettingsError(
+                            "Image Content URI needs to be set, because onnx can only work with file paths"
+                        )
+        # Currently Onnx Runtime only supports one image
+        # Later we will add support for multiple images
+        return images[-1] if len(images) > 0 else None
```

Should we fail explicitly instead of silently ignoring all the other images?